### ISSUE TITLE
Fullscreen event not fired on macos

### ISSUE THREAD

**Issue by nickfujita** [SnosMe/electron-overlay-window#37]
Trying out the demo with TextEdit on MacOS 13.5.1.

When toggling the app to fullscreen like is shown in the video on the original [MacOS support PR](https://github.com/SnosMe/electron-overlay-window/pull/29) the overlay disappears. I put a console log in the [handleFullscreen event handler](https://github.com/SnosMe/electron-overlay-window/blob/master/src/index.ts#L140), but it appears that the event is never emitted. Also placed a console log in the [handler](https://github.com/SnosMe/electron-overlay-window/blob/master/src/index.ts#L179), but only see resize events being fired, not fullscreen.

@hsource was wondering if you have seen this before, or possible are seeing the same on latest MacOS version? (don't have a way to test on older macos versions to confirm whether it used to work or not)

**Comment by hsource** [comment]
Yeah, the fullscreen detection was always a bit of a hack. I'm pretty sure there are better approaches, so you may want to rebuild and experiment with it if possible!

The function to really modify is here: https://github.com/SnosMe/electron-overlay-window/blob/master/src/lib/mac.mm#L570

I really based it off of a Stackoverflow answer which felt like a bit of a hack. I do wonder if there are newer APIs or approaches that are less of a hack.

**Comment by nickfujita** [comment]
Thanks for the response and the code reference. Will look into modifying this. 🙏🏼

**Comment by nickfujita** [comment]
Thanks for the response and the code reference. Will look into modifying this. 🙏🏼
